### PR TITLE
Use sdl2 on mac platform

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

https://github.com/cocos/cocos-engine-external/pull/400
https://github.com/cocos/cocos-engine/issues/15893

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/engine/Engine.cpp [hunk 1]
```diff
@@ -274,7 +274,7 @@ void Engine::tick() {
 
         // iOS/macOS use its own fps limitation algorithm.
         // Windows for Editor should not sleep,because Editor call tick function synchronously
-#if (CC_PLATFORM == CC_PLATFORM_ANDROID || (CC_PLATFORM == CC_PLATFORM_WINDOWS && !CC_EDITOR) || CC_PLATFORM == CC_PLATFORM_OHOS || CC_PLATFORM == CC_PLATFORM_OPENHARMONY) || (defined(CC_SERVER_MODE) && (CC_PLATFORM == CC_PLATFORM_MAC_OSX))
+#if (CC_PLATFORM == CC_PLATFORM_ANDROID || (CC_PLATFORM == CC_PLATFORM_WINDOWS && !CC_EDITOR) || CC_PLATFORM == CC_PLATFORM_OHOS || CC_PLATFORM == CC_PLATFORM_OPENHARMONY || CC_PLATFORM == CC_PLATFORM_MAC_OSX)
```
Comment (dumganhar) [inline native/cocos/engine/Engine.cpp]: Use `CC_PLATFORM_MACOS` instead of `CC_PLATFORM_MAC_OSX`

Comment (qiuguohua) [reply native/cocos/engine/Engine.cpp]: OK

File: native/cocos/platform/mac/modules/SystemWindow.mm [hunk 2]
```diff
@@ -24,41 +23,74 @@ of this software and associated engine source code (the "Software"), a limited,
...
+#include "base/Log.h"
+#include "base/Macros.h"
+
+// SDL headers
+#include <functional>
+#include "SDL2/SDL.h"
+#include "SDL2/SDL_main.h"
+#include "SDL2/SDL_syswm.h"
+#include "engine/EngineEvents.h"
+#include "platform/SDLHelper.h"
```
Comment (minggo) [inline native/cocos/platform/mac/modules/SystemWindow.mm:36]: Add header files as the sequences:
- system related header files
- 3rd libs
- engine related

File: native/external-config.json [hunk 3]
```diff
@@ -3,6 +3,6 @@
         "type": "github",
         "owner": "cocos-creator",
         "name": "engine-native-external",
-        "checkout": "v3.8.1-10"
+        "checkout": "v3.8.1-16"
```
Comment (minggo) [inline native/external-config.json]: It is v3.8.1-13

**Comment by minggo** [comment]
Should support macOS arm64.

**Review by minggo — APPROVED** [review]

File: native/cocos/platform/mac/modules/SystemWindow.h [hunk 4]
```diff
@@ -24,51 +24,49 @@
...
-
+    NSWindow* getNSWindow() const;
 private:
-    bool _pointerLock{false};
-    float _lastMousePosX{0.0F};
-    float _lastMousePosY{0.0F};
-    int32_t _width{0};
-    int32_t _height{0};
+    SDL_Window* getSDLWindow() const { return _window; }
+    void InitWindowProperty(SDL_Window* window, const char *title, int x, int y, int w, int h);
```
Comment (dumganhar) [inline native/cocos/platform/mac/modules/SystemWindow.h]: lowercase for function name.

**Review by dumganhar — APPROVED** [review]

**Pull request merged**